blockchain/state_transition.go

```diff
@@ -335,6 +338,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		st.state.PrepareAccessList(rules, msg.ValidatedSender(), msg.ValidatedFeePayer(), st.evm.Coinbase, msg.To(), vm.ActivePrecompiles(rules))
 	}
 
+	// Check whether the init code size has been exceeded.
+	if rules.IsMantle && msg.To() == nil && len(st.data) > params.MaxInitCodeSize {
+		return nil, fmt.Errorf("%w: code size %v limit %v", ErrMaxInitCodeSizeExceeded, len(st.data), params.MaxInitCodeSize)
+	}
```

Comment:
This is a static validation and it can be applied in the txpool validation as well as in the tx execution process. Then, we can prevent including revert transactions in a block in advance. 
It seems that the following line does the same validation. Please check and applied it in Klaytn's tx validation logic. https://github.com/ethereum/go-ethereum/blame/7f756dc1185d7f1eeeacb1d12341606b7135f9ea/core/txpool/validation.go#L70-L72

Reply:
Thank you. Added the tx.data length check logic just after the ethTxType validation. https://github.com/klaytn/klaytn/pull/1888/commits/75f37abe4b4530a35239d71e7a01e7676c0328ca